### PR TITLE
Show temporary message while sending it

### LINKED ISSUE #271: Add temporary message in the list while sending it

### PR DESCRIPTION
**Pull request by rafaelks**
Closes #271

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatDataController.swift [hunk 1]
```diff
@@ -128,7 +128,16 @@ final class ChatDataController {
...
+        for (index, _) in data.enumerated() {
+            var obj = data[index]
 
+            if obj.message?.identifier == message.identifier {
+                obj.message = message
+                return obj.indexPath.row
+            }
+        }
+
+        return -1
```
Comment (rakaramos) [inline Rocket.Chat/Controllers/Chat/ChatDataController.swift]: Why don't you return `Int?`

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 2]
```diff
@@ -253,17 +253,40 @@ final class ChatViewController: SLKTextViewController {
     // MARK: Message
 
     fileprivate func sendMessage() {
-        guard let message = textView.text else { return }
+        guard let messageText = textView.text else { return }
 
-        if message.characters.count == 0 {
+        if messageText.characters.count == 0 {
```
Comment (rakaramos) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: why not?
```
guard let messageText = textView.text, messageText.characters.count > 0 else { return }
```

File: Rocket.Chat/Controllers/Chat/ChatDataController.swift (hunk 1 above)
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatDataController.swift]: What you mean? `return Int`?

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift (hunk 2 above)
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Sounds good, will change it.

File: Rocket.Chat/Controllers/Chat/ChatDataController.swift (hunk 1 above)
Comment (rakaramos) [inline Rocket.Chat/Controllers/Chat/ChatDataController.swift]: Optional Int

**Pull request merged**